### ISSUE TITLE
fatal errors are not logged on startup

### ISSUE THREAD

**Issue by grexe** [arangodb/arangodb#3634]
When the configuration is invalid (see below) and `arangod` refuses to start, you don't see the error in the log `/var/lib/arangodb3/arangod.log`, but only when you start the daemon by hand, you can see the cause, e.g.:
```
content of 'ENGINE' file '/var/lib/arangodb3/ENGINE' and command-line/configuration option
```
I'm running ArangoDB 3.2.5 in a docker container (Docker version 17.05.0-ce, build 89658be) with Ubuntu 14.04 and `/var/lib` is a host mounted volume, this could be an issue with docker caching a write but I doubt it.

The error can re reproduced by supplying an invalid configuration, e.g. by writing a different storage engine into the '/var/lib/arangodb3/ENGINE' file.

**Comment by dothebart** [comment]
when you use docker, you should be able to see `docker logs <containerid>` which is the stdout of the process - thats the way docker intends it.

**Comment by dothebart** [comment]
@grexe did the above help you?

**Comment by grexe** [comment]
thanks @dothebart that helped, but it shows the real cause of the issue: ArangoDB logs this error only to `STDOUT` but not to the log file in `/var/lib/arangodb3`.

**Comment by dothebart** [comment]
you can add commandline options to the arangod started arangodb docker container:

    docker run -e ARANGO_RANDOM_ROOT_PASSWORD=1 -p 8529:8529 -d arangodb arangod --log.file /var/log/arangodb3/arangod.log

However, we currently have an issue with not enough logging showing up in the foreground, so `docker logs` won't tell you the full picture.

**Comment by dothebart** [comment]
Logging-flush was fixed with ArangoDB 3.3.2; Closing this issue now.
